types/async/index.d.ts

```diff
@@ -246,10 +247,10 @@ export const eachOf: typeof forEachOf;
 export const eachOfSeries: typeof forEachOf;
 export const eachOfLimit: typeof forEachOfLimit;
 export function map<T, R, E = Error>(arr: T[] | IterableIterator<T> | Dictionary<T>, iterator: AsyncResultIterator<T, R, E>, callback: AsyncResultArrayCallback<R, E>): void;
-export function map<T, R, E = Error>(arr: T[] | IterableIterator<T> | Dictionary<T>, iterator: AsyncResultIterator<T, R, E>): Promise<R[]>;
+export function map<T, R, E = Error>(arr: T[] | IterableIterator<T> | Dictionary<T>, iterator: AsyncResultIteratorPromise<T, R>): Promise<R[]>;
```

Comment:
Please keep the old type definition alongside your new one, since both are valid. There should actually be four definitions:

1. `map<T, R, E = Error>(arr: T[] | IterableIterator<T> | Dictionary<T>, iterator: AsyncResultIterator<T, R, E>, callback: AsyncResultArrayCallback<R, E>): void`
2. `map<T, R, E = Error>(arr: T[] | InterableIterator<T> | Dictionary<T>, iterator: AsyncResultIterator<T, R, E>): Promise<R[]>`
3. `map<T, R, E = Error>(arr: T[] | IterableIterator<T> | Dictionary<T>, iterator: AsyncResultIteratorPromise<T, R>, callback: AsyncResultArrayCallback<R, E>): void`
4. `map<T, R, E = Error>(arr: T[] | InterableIterator<T> | Dictionary<T>, iterator: AsyncResultIteratorPromise<T, R>): Promise<R[]>`

Please update the other functions below as well.

Reply:
Added 4 combinations;
+ callback / callback
+ callback / promise
+ promise / callback
+ callback / callback